packages/core/src/libraries/domain.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  type CloudflareData,
+  type Domain,
+  type DomainDnsRecords,
+  DomainStatus,
+} from '@logto/schemas';
+
+import type Queries from '#src/tenants/Queries.js';
+import SystemContext from '#src/tenants/SystemContext.js';
+import assertThat from '#src/utils/assert-that.js';
+import { getCustomHostname, createCustomHostname } from '#src/utils/cloudflare/index.js';
+
+export type DomainLibrary = ReturnType<typeof createDomainLibrary>;
+
+const getDomainStatusFromCloudflareData = (data: CloudflareData): DomainStatus => {
+  if (data.status === 'active' && data.ssl.status === 'active') {
+    return DomainStatus.Active;
+  }
+
+  if (data.status === 'active' && data.ssl.status !== 'active') {
+    return DomainStatus.PendingSsl;
+  }
+
+  if (data.status !== 'active') {
+    return DomainStatus.PendingVerification;
+  }
+
+  return DomainStatus.Error;
```

Comment:
use switch?

Reply:
It depends on 2 variables, so switch may not be easy to use?

Reply:
looks like cleaner?

```ts
switch (data.status) {
  case 'active':
    return data.ssl.status === 'active' ? DomainStatus.Active : DomainStatus.PendingSsl;
  default:
    return DomainStatus.PendingVerification;
}
```

i didn't see which case will fall into DomainStatus.Error though

Reply:
Great! Updated. The "error" state is reserved.